source/examples/generated/TransactionsTest.snippet.transaction.kt

```diff
@@ -0,0 +1,39 @@
+// Set up the MongoDB client and get the collection
+suspend fun performTransaction(client: MongoClient) {
+    client.startSession().use { session ->
+        session.startTransaction() // Start the transaction
+        try {
+            val database = client.getDatabase("bank")
+
+            val savingsColl = database.getCollection<SavingsAccount>("savings_accounts")
+            savingsColl.findOneAndUpdate(
+                session,
+                SavingsAccount::accountId eq "9876",
+                inc(SavingsAccount::amount, -100)
+            )
+         
+            val checkingColl = database.getCollection<CheckingAccount>("checking_accounts")
+            checkingColl.findOneAndUpdate(
+                session,
+                CheckingAccount::accountId eq "9876",
+                inc(CheckingAccount::amount, 100)
+            )
+            // Commit the transaction
+            session.commitTransaction()
+            println("Transaction committed.")
+        } catch (error: Exception) {
+            println("An error occurred during the transaction: ${error.message}")
+            // Abort the transaction
+            session.abortTransaction() 
+        }
+    }
+}
+
+data class SavingsAccount(val accountId: String, val amount: Int)
+data class CheckingAccount(val accountId: String, val amount: Int)
```

Comment:
I'd lift these class definitions out of this example and introduce them above the code example proper.

I'd say after the 3 item list, you can say something like "This example uses the following {+language+} data classes to model its documents:" and then have these two lines above the bulk of the example. See [here](https://preview-mongodbmcmorisi.gatsbyjs.io/kotlin-sync/master/read/retrieve/) for something similar.

Reply:
That makes sense. Will update!